backend/tracim_backend/views/search_api/schemas.py

```diff
@@ -0,0 +1,245 @@
+    def __init__(
+        self,
+        size: int = 10,
+        page_nb: int = 1,
+        search_string: str = "",
+        content_types: typing.Optional[typing.List[str]] = None,
+        show_deleted: int = 0,
+        show_archived: int = 0,
+        show_active: int = 1,
+    ):
+        self.search_string = search_string
+        self.size = size
+        self.page_nb = page_nb
+
+        if not content_types:
+            self.content_types = content_type_list.restricted_allowed_types_slug()
+
+        self.show_deleted = bool(show_deleted)
+        self.show_archived = bool(show_archived)
+        self.show_active = bool(show_active)
+
+
+class ContentSearchFilterQuerySchema(marshmallow.Schema):
+    search_string = StrippedString(
+        example="test", description="just a search string", required=False
+    )
+    size = marshmallow.fields.Int(
+        required=False, default=10, validate=strictly_positive_int_validator
+    )
+    page_nb = marshmallow.fields.Int(
+        required=False, default=1, validate=strictly_positive_int_validator
+    )
+    content_types = StrippedString(
+        required=False, validate=regex_string_as_list_of_string, description="content_types to show"
+    )
+    show_archived = marshmallow.fields.Int(
+        example=0,
+        default=0,
+        description="if set to 1, then show archived contents."
+        " Default is 0 - hide archived content",
+        validate=bool_as_int_validator,
+    )
+    show_deleted = marshmallow.fields.Int(
+        example=0,
+        default=0,
+        description="if set to 1, then show deleted contents."
+        " Default is 0 - hide deleted content",
+        validate=bool_as_int_validator,
+    )
+    show_active = marshmallow.fields.Int(
+        example=1,
+        default=1,
+        description="if set to 1, then show active contents. "
+        "Default is 1 - show active content."
+        " Note: active content are content "
+        "that is neither archived nor deleted. "
+        "The reason for this parameter to exist is for example "
+        "to allow to show only archived documents",
+        validate=bool_as_int_validator,
+    )
+
+    # @post_load
+    # def make_search_content_filter(self, data: typing.Dict[str, typing.Any]) -> object:
+    # return ContentSearchFilterQuery(
+    # search_string=data["search_string"],
+    # size=data["size"],
+    # page_nb=data["page_nb"],
+    # content_types=data["content_types"],
+    # show_archived=data["show_archived"],
+    # show_deleted=data["show_deleted"],
+    # show_active=data["show_active"],
+    # )
+
+
+class EnumField(marshmallow.fields.Field):
+    def __init__(self, enum_cls: typing.Type[Enum], **kwargs):
+        super().__init__(**kwargs)
+        self._enum = enum_cls
+
+    def _deserialize(self, value: str, *arg: typing.Any, **kwargs: typing.Any):
+        for val in self._enum.__members__.values():
+            if value == val.value:
+                return val
```

Comment:
You can simply construct the enum instead of looping over the existing values, e.g.:
```python
return self._enum(val)
```
If the value is not an enum it will raise `ValueError` which should probably be re-raised as a `ValidationError`.

Reply:
Nice, I searched for a way to do this and did not find it.